### DISCUSSION TITLE
Calendar styles not loaded on Vercel cloud

### DISCUSSION THREAD

**Discussion by PrimozRome** [gpbl/react-day-picker#2493] [Support]
For some reason my styles are not loaded in production (Vercel). They are wroking fine locally:

<img width="1573" alt="Screenshot 2024-09-27 at 09 57 46" src="https://github.com/user-attachments/assets/0d6e9feb-0563-4a75-ba77-3a55a7a5576c">

But in production on Vercel not. Check here the calendar: https://balkancampers.vercel.app/campervans/bibita

<img width="1573" alt="Screenshot 2024-09-27 at 09 55 39" src="https://github.com/user-attachments/assets/1a97b59b-e624-4c16-8d55-0f7c6de568f4">

I am importing all styles in my NextJS's `_app.js` file like this:

```
import "react-day-picker/style.css";
import "../public/styles/index.css";

import { GoogleTagManager } from "@next/third-parties/google";
import Layout from "@/components/layout/Layout.js";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "../theme/index.js";

function MyApp({ Component, pageProps }) {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <Layout>
        <Component {...pageProps} />
      </Layout>
      <GoogleTagManager gtmId="GTM-5VRD865" />
    </ChakraProvider>
  )
}

export default MyApp;

```

I was following this guide https://daypicker.dev/docs/styling. Everything works in local dev. environment but when deployed to Vercel the styles are not loaded.

**Comment by Tevuu** [comment]
Hi! I have exactly the same problem. Locally all styles work correctly, but as soon as you install the site, everything immediately crashes. This problem also works if you clone the repository and run the project, all styles also crash.

**Comment by gpbl** [comment]
@PrimozRome, could you check if the issue occurs when running the built project locally?

Vercel might not be building CSS imported from a package. You could try copying the CSS from node_modules to a relative directory to see if that resolves it.

This looks like a Vercel issue, so please reach out to their support or provide us with a way to quickly reproduce the problem. Unfortunately, I can’t spend much time on issues without a clear reproduction path...

**Reply by PrimozRome** [reply]
I copied CSS from node_modules into my public/styles -> same result. Working locally, not working on Vercel. Very strange! Especially becasue I load another CSS from the same path and that one is loaded!

Here is the CSS link: https://balkancampers.vercel.app/styles/daypicker.css

This is how I include it in `_app.js`

```
import "../public/styles/daypicker.css";
import "../public/styles/index.css";
```

I built application locally - same situation

How would you like me to get you a clear reproduction path? I know you are busy but I can send you the code via github?

BTW, Vercel is building CSS imported from a package... I have used other packages, no problems. I will try to reach out to Vercel if they can help.

**Reply by PrimozRome** [reply]
I did some additional tests. 

For some reason all the the `rdp-` CSS classes that Day picker module is using aren't added into the CSS file when app is built. 

If I use one of the `rdp-` classes from the module in my markup, that only that class is added into the built CSS file. 

So at this moment I have zero clue why is that!!

**Reply by PrimozRome** [reply]
I think it is the same situation on the Code Sandbox. I did a quick demo with default NextJS template. Here is the link:

https://codesandbox.io/p/devbox/s6g5yk

No styles are loaded here as well...

**Reply by PrimozRome** [reply]
**ANOTHER UPDATE**

I found a solution that works. Loaded Day Picker CSS file from [jsDeliver](https://www.jsdelivr.com/package/npm/react-day-picker) in `_document.js` file using `<link />`. Not the ideal solution but it works.

```
import Document, { Html, Head, Main, NextScript } from "next/document";
import { ColorModeScript } from '@chakra-ui/react'
import theme from '/theme'

export default class MyDocument extends Document {
  render() {
    return (
      <Html lang="sl">
        <Head>
          <link rel="preconnect" href="https://fonts.googleapis.com" />
          <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
          <link href="https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,100..900;1,100..900&display=swap" rel="stylesheet" />
          <link href="https://cdn.jsdelivr.net/npm/react-day-picker@9.1.3/src/style.min.css" rel="stylesheet" />
        </Head>
        <body>
          <ColorModeScript initialColorMode={theme.config.initialColorMode} />
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```

**Comment by gpbl** [comment]
@PrimozRome thanks for your updates and for the code sandbox. I couldn't replicate the issue there either;

https://codesandbox.io/p/sandbox/day-picker-next-js-p4zzzy

It looks like there's something in the CSS that Vercel doesn't like.

Could it be the [css variables](https://github.com/gpbl/react-day-picker/blob/60d071b866efd76f10dfc5b8762f8f93e0953485/src/style.css#L1-L64)?

> I copied CSS from node_modules into my public/styles -> same result. 

Interesting...

- Try by finding or removing the code that could break the build. 
- Investigate the built CSS: what is in there related to daypicker
- Try to use the css module: https://daypicker.dev/docs/styling#importing-the-css-module
